src/components/navigation-control.js

```diff
@@ -40,7 +44,9 @@ export default class NavigationControl extends Component {
   _updateViewport(opts) {
     const {viewport} = this.context;
     const mapState = new MapState(Object.assign({}, viewport, opts));
-    this.props.onChangeViewport(mapState.getViewportProps());
+    // TODO(deprecate): remove this check when `onChangeViewport` gets deprecated
+    const onViewportChange = this.props.onChangeViewport || this.props.onViewportChange;
```

Comment:
Since we already have a table with prop mappings in dep-warn, couldn't we have a function that uses that table and applies the value of old props if new ones are not supplied, instead of duplicating that logic here?